test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -1328,8 +1330,14 @@ TEST_F(PickFirstTest, PendingUpdateAndSelectedSubchannelFails) {
           "Phase 2: Resolver update pointing to remaining "
           "(not started) servers.");
   response_generator.SetNextResolution(GetServersPorts(1 /* start_index */));
+  // Add hold before connection attempt to ensure RPCs will be sent to first
+  // server. Otherwise, pending subchannel list might already have gone into
+  // TRANSIENT_FAILURE due to hitting the end of the server list by the time
+  // we check the state.
+  hold->Wait();
   // RPCs will continue to be sent to the first server.
   CheckRpcSendOk(DEBUG_LOCATION, stub);
+  hold->Resume();
```

Comment:
I don't think we want to resume this until after we check the channel's state on line 1353.  Or to say this another way, I think this should replace the `StartServer(1)` call on line 1358.

Also, I think we can probably change the check on line 1353 to accept only CONNECTING, not TRANSIENT_FAILURE.  I suspect that this PR will also fix the flake described in the TODO on line 1349 (and that TODO comment can go away now).

Reply:
Done